Changes drawer navigation from drawer to accordion

eduludi opened the pull request:
This replaces the "sub-drawer" behaviour from drawer menu on mobile, with an accordion-like menu.

* Updates some `.c-drawer` styles: 
  - Subnav arrows points up and down in mobile
  - Increases padding in first and last child items
  - Disables user text selection of menu items (may happen on desktop when user makes a double-click on the arrow)
  - Removes some no longer used styles
  - Updates item and link colors
* Updates `$('.c-primary-nav__arrow').on(click)` event
* Removes placeholder `c-drawer__subnav` from  `drawer-navigation.twig` (only used to hold the cloned menu item)

eduludi committed:
Changes drawer navigation from drawer to accordion

- Updates some `.c-drawer` styles
- Updates `$('.c-primary-nav__arrow').on(click)` event
- Removes placeholder `c-drawer__subnav` from  `drawer-navigation.twig`

designerbrent commented on source/_patterns/01-molecules/navigation/drawer-navigation.twig:
```diff
@@ -14,7 +14,6 @@
     <div class="c-drawer__nav">
       <div class="c-drawer__nav-primary">
         {% include '@molecules/navigation/primary-navigation.twig' %}
-        <ul class="c-drawer__subnav u-theme--background-color--darker"></ul>
```
@eduludi Is it possible to add a class or CSS change that would make this not necessary to remove? Since we have to coordinate with the update of the Wordpress theme, it would be helpful to be able to leave this line in for at least the next month or so to provide some overlap without messing up the existing installations.

eduludi replied:
It wasn't necessary to remove but, as the JS part to make it useful is not longer there, I just did it. 

Now I rolled back that change and only two files (CSS and JS) are being modified by this PR.

designerbrent replied:
The problem I'm seeing is that if you don't remove it, the menu is still unusable on the phone, as it slides out a blank drawer... like before, but with nothing in it. Not sure if that can be hidden?

eduludi committed:
Sets back .c-drawer__subnav in drawer-navigation.twig

designerbrent commented:
This is still in need of some work, as the video shows, the overlays are still visible with the most recent code. 

[🎥 Test on mobile](https://www.dropbox.com/s/b565mfr9e8048hg/ALPS-mobile-menu-test.MP4?dl=0)

designerbrent commented:
When the menu opens, the indicator needs to be flipped over.
![2020-07-22 at 5 07 PM](https://user-images.githubusercontent.com/132864/88228942-ee38e980-cc3d-11ea-9014-ce6ca76d48fa.png)

eduludi committed:
Removes ul.c-drawer__subnav from drawer-navigation template

eduludi committed:
Fixes issue with hover in drawer items after resize

- It was being triggered in small resolutions after resizing from a larger one
- This caused expand arrows to point in the wrong direction

eduludi commented:
> This is still in need of some work, as the video shows, the overlays are still visible with the most recent code.
> 
> [🎥 Test on mobile](https://www.dropbox.com/s/b565mfr9e8048hg/ALPS-mobile-menu-test.MP4?dl=0)

This was caused by some CSS selector related to `.c-drawer__subnav`, so it will be required to remove the element from template (as was my original PR). 

We can remove related CSS in a future version, as it still may be used by some implementations consuming the `latest` CSS.

> When the menu opens, the indicator needs to be flipped over.
> ![2020-07-22 at 5 07 PM](https://user-images.githubusercontent.com/132864/88228942-ee38e980-cc3d-11ea-9014-ce6ca76d48fa.png)

I was only able to replicate this when resized the browser from a `desktop`/` tablet` to a `mobile` resolution. And this was caused because existing script was testing for window's width only at the beginning.
